pkg/server/domain/service/workflow.go

```diff
@@ -499,114 +373,76 @@ func (w *workflowServiceImpl) setRecordToTerminated(ctx context.Context, appPrim
+			klog.Errorf("failed to load the context values of the application %s:%s", app.Name, err.Error())
 		}
-		return err
-	}
-	var revision = &model.ApplicationRevision{AppPrimaryKey: appPrimaryKey, Version: record.RevisionPrimaryKey}
-	if err := w.Store.Get(ctx, revision); err != nil {
-		if errors.Is(err, datastore.ErrRecordNotExist) {
-			return bcode.ErrApplicationRevisionNotExist
-		}
-		return err
-	}
-
-	if workflowContext != nil {
-		record.ContextValue = workflowContext
+		record.ContextValue = cm.Data
 	}
 
-	if app.Status.Workflow != nil {
-		if app.Status.Workflow.AppRevision != record.Name {
-			klog.Warningf("the app(%s) revision is not match the record(%s), try next time..", app.Name, record.Name)
-			return nil
+	stepStatus := make(map[string]*model.WorkflowStepStatus, len(status.Steps))
+	stepAlias := make(map[string]string)
+	for _, step := range record.Steps {
+		stepAlias[step.Name] = step.Alias
+		for _, sub := range step.SubStepsStatus {
+			stepAlias[sub.Name] = sub.Alias
 		}
-		status := app.Status.Workflow
-		record.Status = string(status.Phase)
-		record.Message = status.Message
-		record.Mode = status.Mode
-
-		if cb := app.Status.Workflow.ContextBackend; cb != nil && workflowContext == nil && cb.Namespace != "" && cb.Name != "" {
-			var cm corev1.ConfigMap
-			if err := w.KubeClient.Get(ctx, types.NamespacedName{Namespace: cb.Namespace, Name: cb.Name}, &cm); err != nil {
-				klog.Errorf("failed to load the context values of the application %s:%s", app.Name, err.Error())
-			}
-			record.ContextValue = cm.Data
+	}
+	for _, step := range status.Steps {
+		stepStatus[step.Name] = &model.WorkflowStepStatus{
+			StepStatus:     convert.FromCRWorkflowStepStatus(step.StepStatus, stepAlias[step.Name]),
+			SubStepsStatus: make([]model.StepStatus, 0),
 		}
-
-		stepStatus := make(map[string]*model.WorkflowStepStatus, len(status.Steps))
-		stepAlias := make(map[string]string)
-		for _, step := range record.Steps {
-			stepAlias[step.Name] = step.Alias
-			for _, sub := range step.SubStepsStatus {
-				stepAlias[sub.Name] = sub.Alias
-			}
+		for _, sub := range step.SubStepsStatus {
+			stepStatus[step.Name].SubStepsStatus = append(stepStatus[step.Name].SubStepsStatus, convert.FromCRWorkflowStepStatus(sub, stepAlias[sub.Name]))
 		}
-		for _, step := range status.Steps {
-			stepStatus[step.Name] = &model.WorkflowStepStatus{
-				StepStatus:     convert.FromCRWorkflowStepStatus(step.StepStatus, stepAlias[step.Name]),
-				SubStepsStatus: make([]model.StepStatus, 0),
-			}
-			for _, sub := range step.SubStepsStatus {
-				stepStatus[step.Name].SubStepsStatus = append(stepStatus[step.Name].SubStepsStatus, convert.FromCRWorkflowStepStatus(sub, stepAlias[sub.Name]))
-			}
-		}
-		for i, step := range record.Steps {
-			if stepStatus[step.Name] != nil {
-				record.Steps[i] = *stepStatus[step.Name]
-			}
+	}
+	for i, step := range record.Steps {
+		if stepStatus[step.Name] != nil {
+			record.Steps[i] = *stepStatus[step.Name]
```

Comment:
There is a bug if the step type is `step-group`. The step status only includes the finished or running steps. This overlay action leads to missing the no-running steps. 

/cc @wangyikewxgm

Reply:
Actually I've fixed this with `else`.

Reply:
Oh, fixed by the vela core?